### ISSUE TITLE
Moved from QGIS 2.18 to 3.8, Views are not valid nor cannot be loaded and map not working due to Key

### ISSUE THREAD

**Issue by camanicnack** [qgis/QGIS#45557]
### What is the bug or the crash?

My company recently went from QGIS 2.18 to QGIS 3.8.  When we moved, our map is currently working correctly, especially when it goes to WMTS, we are able to see every layer of the map we go to the live website.  However, when we either add a new view or replace another view with an updated version it is simply not working.   We either get two results...  Either we can see the view in the map of the QGIS software, but when it translates to the WMTS image (which what we mean by going to the live website) it does not show the view at all.  Or... It gives us an "Invalid Error" within the logs and say that we cannot put the particular view in.    
It seems as if the issue could be a key problem.  The old views keys are "triannual", but the new view's keys are countyid.  So we tried changing the keys, it keeps on giving us a invalid error and the layer will not go in.   We even tried changing to ssl mode to allow, and then fix the keys as of that way.  QGIS system still not allowing us to change and keep getting invalid error messages in the logs.  We tried to go from the DB Manager and click n dragging over to the layers.  No good.  Try going from the Browser Panel into the layers, again no good.  Error messaages.  Even tried the Database portion, and still not able to get it to work.  

Please any help to have the layer show up on our maps is appreciated.  

### Steps to reproduce the issue

It seems as if the issue could be a key problem.  The old views keys are "triannual", but the new view's keys are countyid.  So we tried changing the keys, it keeps on giving us a invalid error and the layer will not go in.   We even tried changing to ssl mode to allow, and then fix the keys as of that way.  QGIS system still not allowing us to change and keep getting invalid error messages in the logs.  We tried to go from the DB Manager and click n dragging over to the layers.  No good.  Try going from the Browser Panel into the layers, again no good.  Error messaages.  Even tried the Database portion, and still not able to get it to work.

Update:
I did update to now 3.2.
Sorry...  I am kinda trying to figure out QGIS and still new with the QGIS system.   As I said was used to QGIS 2.18 but upgraded to 3.8..   Now I did finally upgrade back to 3.2 and I am still having the issue.   
Main issue is we are having problems with the views from postgres.  The old views the unique key was set to triannual and sslmode set to allow.   However, when I upgraded to 3.2, the when I apply new views as a layer for the map, it is not allowing me to set the unique key to triannaul but to something else.  And... not allowing me to set up the sslmode to allow.  The system keeps giving me an error that the layer is not valid and cannot add.  Again the old layers which are showing the map has this but cannot set the new layers.  Now I can try setting it something else and then it shows the layer within  the app map builder, but when I save it.  It doesn't show the image of that layer within the live map.  It somehow needs that unique key but not allowing me to change the unique key to triannual.  

### Versions

Went from QGIS 2.18 to new version of QGIS 3.8

### Supported QGIS version

- [x] I'm running a supported QGIS version according to the roadmap.

### New profile

- [X] I tried with a new QGIS profile

### Additional context

Any help is appreciated in order to update our maps and be shown on the WMTS image.

**Comment by jef-n** [comment]
The issue is not really clear to me, but 3.8 is EOL anyway.   Can you reproduce the problem with 3.16 (LTR) or 3.20?  If so, please also provide a test case illustrating the problem.

**Comment by camanicnack** [comment]
Sorry...  I am kinda trying to figure out QGIS and still new with the QGIS system.   As I said was used to QGIS 2.18 but upgraded to 3.8..   Now I did finally upgrade back to 3.2 and I am still having the issue.   
Main issue is we are having problems with the views from postgres.  The old views the unique key was set to triannual and sslmode set to allow.   However, when I upgraded to 3.2, the when I apply new views as a layer for the map, it is not allowing me to set the unique key to triannaul but to something else.  And... not allowing me to set up the sslmode to allow.  The system keeps giving me an error that the layer is not valid and cannot add.  Again the old layers which are showing the map has this but cannot set the new layers.  Now I can try setting it something else and then it shows the layer within  the app map builder, but when I save it.  It doesn't show the image of that layer within the live map.  It somehow needs that unique key but not allowing me to change the unique key to triannual.
